src/sentry/static/sentry/app/views/organizationDiscover/result/utils.jsx

```diff
@@ -16,17 +16,38 @@ const CHART_KEY = '__CHART_KEY__';
  *
  * @param {Array} data Data returned from Snuba
  * @param {Object} query Query state corresponding to data
+ * @param {Object} [options] Options object
+ * @param {Boolean} [options.includePercentages] (default: false) Include aggregation's percentages
  * @returns {Array}
  */
-export function getChartData(data, query) {
+export function getChartData(data, query, options = {}) {
   const {fields} = query;
 
+  const totalsBySeries = new Map();
+
+  if (options.includePercentages) {
+    query.aggregations.forEach(aggregation => {
+      totalsBySeries.set(
+        aggregation[2],
+        data.reduce((acc, res) => {
+          acc += res[aggregation[2]];
+          return acc;
+        }, 0)
+      );
+    });
+  }
+
   return query.aggregations.map(aggregation => {
+    const total = options.includePercentages && totalsBySeries.get(aggregation[2]);
```

Comment:
Rather than adding an additional check for includePercentages here, how about doing this inline in line 49 (or extract a function that does both together)

Reply:
L49 is inside of a `map` so you'd be calling `get()` every iteration even though it won't change

Reply:
We only ever chart a single aggregation here so shouldn't we just skip the map and calculating every total anyway?

Reply:
Discover and dashboards will be closely related, I think it's better to have the logic in the same place, no?

Reply:
I mean you can write a new function and place it next to this one (like the by day query?) Discover itself is designed to take a different processing function for each chart type each of which does a specific transformation, so I'd suggest making a new function for your purpose next to this one since it appears to require quite a different format?

Reply:
It'll need a similar format in that it expects an object with `value` and `name` in addition to `percentage`

Reply:
@lynnagara I don't think this is better, but I moved it into its own function here 142d9c5eeb603c38c059d11393f7669e83e28568

Reply:
Thanks. I was hoping we can remove all the options from `getChartData` though not just this particular one and move all the new code to the percentage function - do you think that's feasible?

Reply:
https://github.com/getsentry/sentry/pull/11329/commits/ad9a12e13bfe259daba982dd95c26967f0192594